scripts/run-integration-tests.sh

```diff
@@ -8,36 +8,31 @@
 # script, they are treated as the directories containing the tests to run.
 # When no arguments are provided, all integration tests are executed.
 
-set -e -u
+set -eu
 
 fabric_dir="$(cd "$(dirname "$0")/.." && pwd)"
+cd ${fabric_dir}
 
-cd "$fabric_dir"
+declare -a test_dirs
+test_dirs=($(
+ go list -f '{{if or (len .TestGoFiles | ne 0) (len .XTestGoFiles | ne 0)}}{{println .Dir}}{{end}}' ./... | \
+ grep integration | \
+ sed "s,${fabric_dir},.,g"
+))
 
-dirs=()
-if [ "${#}" -eq 0 ]; then
-  specs=()
-  specs=("$(grep -Ril --exclude-dir=vendor --exclude-dir=scripts "RunSpecs" . | grep integration)")
-  for spec in ${specs[*]}; do
-    dirs+=("$(dirname "${spec}")")
-  done
-else
-  dirs=("$@")
-fi
-
-totalAgents=${SYSTEM_TOTALJOBSINPHASE:-0}   # standard VSTS variables available using parallel execution; total number of parallel jobs running
-agentNumber=${SYSTEM_JOBPOSITIONINPHASE:-0} # current job position
-testCount=${#dirs[@]}
+total_agents=${SYSTEM_TOTALJOBSINPHASE:-0}   # standard VSTS variables available using parallel execution; total number of parallel jobs running
+agent_number=${SYSTEM_JOBPOSITIONINPHASE:-0} # current job position
```

Comment:
Why are you just using '1' as the default since you're effectively doing that in the next block.

Reply:
Back when we originally did this, there was some reason for it related to how the 0th agent handles it. This was an exact copy of the Microsoft example for doing it.

I can't find the doc anymore that tells you why because they deleted the golang example. So I'll remove it and see what happens.